### PR TITLE
Jupyter instance Ansible and Terraform configuration

### PR DESCRIPTION
**Pull request by hevans66**
* Creates terraform configuration for a new jupyter ec2 instance
* Adds cloudflare provier for DNS
* Adds ansible playbook to provision jupyter ec2 instance with the littlest jupyter hub (tljh from now on)
* Adds ansible playbook to provision tljh instance with users and teams and a shared folder

These playbooks do not add a read only folder for examples.

Questions:
* What size should this jupyter instance be? I have no sense of how resource intensive your average jupyter workload is. Space on the root volume, is 10Gb reasonable or are people pulling large files?
* Is there an existing fun naming convention? I'm calling this one "littlehub" but if we have a naming scheme going for other things, like fun proteins or something we should use that.

Note:
This setup does have the issue that data on the root device is stateful. Once customers actually start using blowing we can't simply blow away the instance. I think for now that is fine. With a more nuanced setup we could mount external EBS volumes and put the users files there. Probably not worth spending time on this now, but its something to know.

### CODE REVIEW

File: infrastructure/ansible/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Ansible is used to provision and configure infrastructure on AWS.
+
+# Prerequisites
+
+1. [Install Anible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html) Most likely this is just `python3 -m pip install --user ansible`
+1. This setup assumes the ssh key to access all ec2 instances is in ~/.ssh/setward-dev.pem.
```
Comment (thetechnocrat-dev) [inline infrastructure/ansible/README.md:6]: 2.

**Review by thetechnocrat-dev — APPROVED** [review]
Code looks good, will need some hand holding to test the ansible scripts. I also, didn't see where the plex repo is installed, did I miss that or is that going to be a separate PR?

To answer your questions:
* For compute, a t3.xlarge with 1,000 GB disk seems good (if I did my math correctly that is $201.47 a month). I think number of vCPUs is what will matter most for concurrent notebook users. Our first benchmark dataset was 10GB. 
* We were going with NeuroSci names, we originally called the provide-compute.sh ganglia.sh where ganglia is a decentralized group of neaurons. And, plex is a rift on plexus which is the connective tissue of neurons with other cells.

File: infrastructure/ansible/README.md (hunk 1 above)
Comment (hevans66) [reply infrastructure/ansible/README.md:6]: Github markdown starts the list at whatever the first number is and [subsequent numbers don't matter](https://github.github.com/gfm/#ordered-list). Makes it easier to add numbers later without having to change the entire list.

See the preview of this readme https://github.com/labdao/plex/blob/1cca535cf640de70315b95935342b392c0219868/infrastructure/ansible/README.md

Some people feel strongly about the markdown being sequential though. I'm happy to change it if we feel strongly.

**Comment by hevans66** [comment]
>  didn't see where the plex repo is installed
Had not pushed that yet. Just pushed it if you want to take a look. Also the snippet that symlinks directly to the plex/python folder is there. We can leave that or do something else if you have new notebooks you'd prefer to link to.

> We were going with NeuroSci names
ok, guess I gotta think of something clever

File: infrastructure/ansible/README.md (hunk 1 above)
Comment (thetechnocrat-dev) [reply infrastructure/ansible/README.md:6]: TIL, works for me.

Comment (acashmoney) [reply infrastructure/ansible/README.md:6]: TIL for me too 😮

File: infrastructure/ansible/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+Ansible is used to provision and configure infrastructure on AWS.
+
+# Prerequisites
+
+1. [Install Anible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html) Most likely this is just `python3 -m pip install --user ansible`
```
Comment (acashmoney) [inline infrastructure/ansible/README.md:5]: Ansible

File: infrastructure/ansible/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+We are using an aws plug in that provides dynamic ec2 inventory. Through configuration in `inventory.aws_ec2.yaml` we tell the dynamic inventory mechanism to group instances together based on AWS instance tags. Setting (for example) a Type tag on your aws instances of the same type (in Terraform) will allow you to target the instances you wish you perform tasks on.
+
+# Playbooks
+
+* `provision_jupyter.yaml` Targets `jupyter_notebook` instances and installs [The Littlest Jupyter Hub](https://tljh.jupyter.org/). It does not do any configuration beyond installation.
+* `set_jupyter_users.yaml` Sets the admins, users, and defines access permissions to a team folder 
+
+# The Teams Task
+
+`set_jupyter_users.yaml` has a Create teams task. It loops over a list of hashes of the form to create the juptyter and unix users and set up the home directories. To add a new user simply add a new line to the appropriate team, or add a enw team.
```
Comment (acashmoney) [inline infrastructure/ansible/README.md:28]: new

**Review by acashmoney — APPROVED** [review]
Highlighted a few typos, but looks good to me. Agree with @thetechnocrat-dev , might need a bit of handholding to start

**Pull request merged**